internal/services/dynatrace/dynatrace_monitors_resource.go

```diff
@@ -0,0 +1,288 @@
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", id, err)
+			}
+			if model := resp.Model; model != nil {
+				props := model.Properties
+				identityProps, err := flattenDynatraceIdentity(model.Identity)
+				if err != nil {
+					return fmt.Errorf("flattening identity: %+v", err)
+				}
+				userInfo := metadata.ResourceData.Get("user").([]interface{})
+				monitoringStatus := true
+				if *props.MonitoringStatus == monitors.MonitoringStatusDisabled {
+					monitoringStatus = false
+				}
+
+				state := MonitorsResourceModel{
+					Name:                          id.MonitorName,
+					ResourceGroup:                 id.ResourceGroupName,
+					Location:                      model.Location,
+					MonitoringStatus:              monitoringStatus,
+					MarketplaceSubscriptionStatus: string(*props.MarketplaceSubscriptionStatus),
+					Identity:                      identityProps,
+					PlanData:                      FlattenDynatracePlanData(props.PlanData),
+					UserInfo:                      FlattenDynatraceUserInfo(userInfo),
+				}
+
+				if model.Tags != nil {
+					state.Tags = *model.Tags
+				}
+
+				return metadata.Encode(&state)
+			}
+			return nil
+		},
+	}
+}
+
+func (r MonitorsResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Dynatrace.MonitorsClient
+			id, err := monitors.ParseMonitorID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("deleting %s", *id)
+
+			if resp, err := client.Delete(ctx, *id); err != nil {
+				if !response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("deleting %s: %+v", *id, err)
+				}
+			}
+			return nil
+		},
+	}
+}
+
+func (r MonitorsResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return monitors.ValidateMonitorID
+}
+
+func (r MonitorsResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Dynatrace.MonitorsClient
+			id, err := monitors.ParseMonitorID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state MonitorsResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				props := monitors.MonitorResourceUpdate{
+					Tags: &state.Tags,
+				}
+
+				if _, err := client.Update(ctx, *id, props); err != nil {
+					return fmt.Errorf("updating %s: %+v", *id, err)
+				}
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandDynatraceIdentity(input []identity.ModelSystemAssigned) (*monitors.IdentityProperties, error) {
```

Comment:
From my observation, this is different because in Swagger the definition of the identity properties is slightly different from the standard identity.